lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2693,83 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Signature for [PlatformDispatcher.onViewFocusChange].
+typedef ViewFocusChangeCallback = void Function(ViewFocusEvent viewFocusEvent);
+
+/// An event for the engine to communicate view focus changes back to the app.
+///
+/// This value will be tipically passed to the [PlatformDispatcher.onViewFocusChange]
+/// callback.
+final class ViewFocusEvent {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusEvent({
+    required this.viewId,
+    required this.focusState,
+    required this.focusDirection,
+  });
+
+  /// The id of the [FlutterView] that experienced a focus change.
+  final int viewId;
+
+  /// The current [ViewFocusState] of the [FlutterView] with the given [viewId].
+  final ViewFocusState focusState;
+
+  /// The current [ViewFocusDirection] of the [FlutterView] with the given [viewId].
+  final ViewFocusDirection focusDirection;
+
+  @override
+  String toString() {
+    return 'ViewFocusEvent(viewId: $viewId, focusState: $focusState, focusDirection: $focusDirection)';
+  }
+}
+
+/// Represents the focus state of a given [FlutterView].
+///
+/// This enum will be associated with a [FlutterView] by instantiating a [ViewFocusEvent].
```

Comment:
It's interesting that an event is the only thing that associates a view with its focus state. We might want to provide `FlutterView.focusState` getter that provides a persistent association.

Reply:
We can certainly add something like `FlutterView.focusState` for the sake of ergonomics but given that focus is a single resource thing I'd prefer a higher entity to be the SOT for this (like `FlutterViewManager`). This is closer to the browser in the sense that it gives you `document.activeElement` instead of a DOM node API to query that information.

Reply:
That entity can be the platform dispatcher itself :shrug:

Reply:
SGTM. In that case, let's remove this paragraph for now, until we have a persistent association between views and this enum. `ViewFocusEvent` doesn't seem to be the right thing to maintain this association, since events are transient. The "See also" section should be sufficient to leave breadcrumbs for the user to find out how this enum is used.